[
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315455221",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315455221",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315455221,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ1NTIyMQ==",
    "user": {
      "login": "codecov[bot]",
      "id": 22429695,
      "node_id": "MDM6Qm90MjI0Mjk2OTU=",
      "avatar_url": "https://avatars2.githubusercontent.com/in/254?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/codecov%5Bbot%5D",
      "html_url": "https://github.com/apps/codecov",
      "followers_url": "https://api.github.com/users/codecov%5Bbot%5D/followers",
      "following_url": "https://api.github.com/users/codecov%5Bbot%5D/following{/other_user}",
      "gists_url": "https://api.github.com/users/codecov%5Bbot%5D/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/codecov%5Bbot%5D/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/codecov%5Bbot%5D/subscriptions",
      "organizations_url": "https://api.github.com/users/codecov%5Bbot%5D/orgs",
      "repos_url": "https://api.github.com/users/codecov%5Bbot%5D/repos",
      "events_url": "https://api.github.com/users/codecov%5Bbot%5D/events{/privacy}",
      "received_events_url": "https://api.github.com/users/codecov%5Bbot%5D/received_events",
      "type": "Bot",
      "site_admin": false
    },
    "created_at": "2017-07-14T20:08:06Z",
    "updated_at": "2017-07-14T20:08:06Z",
    "author_association": "NONE",
    "body": "# [Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=h1) Report\n> Merging [#16930](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=desc) into [master](https://codecov.io/gh/pandas-dev/pandas/commit/6000c5b9624fdd8925099f215eba282bfbef87ce?src=pr&el=desc) will **decrease** coverage by `0.01%`.\n> The diff coverage is `100%`.\n\n[![Impacted file tree graph](https://codecov.io/gh/pandas-dev/pandas/pull/16930/graphs/tree.svg?token=eZ4WkYLtcO&src=pr&width=650&height=150)](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree)\n\n```diff\n@@            Coverage Diff             @@\n##           master   #16930      +/-   ##\n==========================================\n- Coverage   90.99%   90.97%   -0.02%     \n==========================================\n  Files         161      161              \n  Lines       49303    49303              \n==========================================\n- Hits        44863    44854       -9     \n- Misses       4440     4449       +9\n```\n\n| Flag | Coverage Δ | |\n|---|---|---|\n| #multiple | `88.74% <100%> (ø)` | :arrow_up: |\n| #single | `40.19% <0%> (-0.07%)` | :arrow_down: |\n\n| [Impacted Files](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree) | Coverage Δ | |\n|---|---|---|\n| [pandas/io/html.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2lvL2h0bWwucHk=) | `84.85% <100%> (ø)` | :arrow_up: |\n| [pandas/io/gbq.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2lvL2dicS5weQ==) | `25% <0%> (-58.34%)` | :arrow_down: |\n| [pandas/core/frame.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2NvcmUvZnJhbWUucHk=) | `97.71% <0%> (-0.1%)` | :arrow_down: |\n\n------\n\n[Continue to review full report at Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=continue).\n> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)\n> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`\n> Powered by [Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=footer). Last update [6000c5b...7688010](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).\n"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315455252",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315455252",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315455252,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ1NTI1Mg==",
    "user": {
      "login": "codecov[bot]",
      "id": 22429695,
      "node_id": "MDM6Qm90MjI0Mjk2OTU=",
      "avatar_url": "https://avatars2.githubusercontent.com/in/254?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/codecov%5Bbot%5D",
      "html_url": "https://github.com/apps/codecov",
      "followers_url": "https://api.github.com/users/codecov%5Bbot%5D/followers",
      "following_url": "https://api.github.com/users/codecov%5Bbot%5D/following{/other_user}",
      "gists_url": "https://api.github.com/users/codecov%5Bbot%5D/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/codecov%5Bbot%5D/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/codecov%5Bbot%5D/subscriptions",
      "organizations_url": "https://api.github.com/users/codecov%5Bbot%5D/orgs",
      "repos_url": "https://api.github.com/users/codecov%5Bbot%5D/repos",
      "events_url": "https://api.github.com/users/codecov%5Bbot%5D/events{/privacy}",
      "received_events_url": "https://api.github.com/users/codecov%5Bbot%5D/received_events",
      "type": "Bot",
      "site_admin": false
    },
    "created_at": "2017-07-14T20:08:15Z",
    "updated_at": "2017-07-21T15:44:22Z",
    "author_association": "NONE",
    "body": "# [Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=h1) Report\n> Merging [#16930](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=desc) into [master](https://codecov.io/gh/pandas-dev/pandas/commit/4efe6560e07f28de6a1834fa90e31cef31b0fb18?src=pr&el=desc) will **decrease** coverage by `0.04%`.\n> The diff coverage is `100%`.\n\n[![Impacted file tree graph](https://codecov.io/gh/pandas-dev/pandas/pull/16930/graphs/tree.svg?height=150&src=pr&token=eZ4WkYLtcO&width=650)](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree)\n\n```diff\n@@            Coverage Diff             @@\n##           master   #16930      +/-   ##\n==========================================\n- Coverage   91.05%   91.01%   -0.05%     \n==========================================\n  Files         161      161              \n  Lines       49350    49350              \n==========================================\n- Hits        44936    44915      -21     \n- Misses       4414     4435      +21\n```\n\n| Flag | Coverage Δ | |\n|---|---|---|\n| #multiple | `88.78% <100%> (-0.03%)` | :arrow_down: |\n| #single | `40.26% <0%> (-0.07%)` | :arrow_down: |\n\n| [Impacted Files](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree) | Coverage Δ | |\n|---|---|---|\n| [pandas/io/html.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2lvL2h0bWwucHk=) | `84.85% <100%> (ø)` | :arrow_up: |\n| [pandas/io/gbq.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2lvL2dicS5weQ==) | `25% <0%> (-58.34%)` | :arrow_down: |\n| [pandas/plotting/\\_converter.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL3Bsb3R0aW5nL19jb252ZXJ0ZXIucHk=) | `63.23% <0%> (-1.82%)` | :arrow_down: |\n| [pandas/core/frame.py](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=tree#diff-cGFuZGFzL2NvcmUvZnJhbWUucHk=) | `97.75% <0%> (-0.1%)` | :arrow_down: |\n\n------\n\n[Continue to review full report at Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=continue).\n> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)\n> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`\n> Powered by [Codecov](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=footer). Last update [4efe656...c01caaf](https://codecov.io/gh/pandas-dev/pandas/pull/16930?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).\n"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315495759",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315495759",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315495759,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NTc1OQ==",
    "user": {
      "login": "3553x",
      "id": 25621909,
      "node_id": "MDQ6VXNlcjI1NjIxOTA5",
      "avatar_url": "https://avatars0.githubusercontent.com/u/25621909?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/3553x",
      "html_url": "https://github.com/3553x",
      "followers_url": "https://api.github.com/users/3553x/followers",
      "following_url": "https://api.github.com/users/3553x/following{/other_user}",
      "gists_url": "https://api.github.com/users/3553x/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/3553x/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/3553x/subscriptions",
      "organizations_url": "https://api.github.com/users/3553x/orgs",
      "repos_url": "https://api.github.com/users/3553x/repos",
      "events_url": "https://api.github.com/users/3553x/events{/privacy}",
      "received_events_url": "https://api.github.com/users/3553x/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:22:04Z",
    "updated_at": "2017-07-15T00:22:04Z",
    "author_association": "CONTRIBUTOR",
    "body": "Regarding adding the test.\r\n\r\nI have trouble creating a test that reproduces the issue.\r\nThe following code snippet works fine when I run `pytest test_html.py::test_importcheck_thread_safety` but as soon as I run `pytest test_html.py` the test always passes. Neither of the threads throw an exception in that case.\r\n```python\r\ndef get_html_file():\r\n    filename = os.path.join(DATA_PATH, 'valid_markup.html')\r\n    read_html(filename)\r\n\r\n\r\nclass ErrorThread(threading.Thread):\r\n    def run(self):\r\n        try:\r\n            threading.Thread.run(self)\r\n        except Exception as e:\r\n            self.err = e\r\n        else:\r\n            self.err = None\r\n\r\ndef test_importcheck_thread_safety():\r\n    t1 = ErrorThread(target = get_html_file)\r\n    t2 = ErrorThread(target = get_html_file)\r\n    t1.start()\r\n    t2.start()\r\n    while(t1.is_alive() or t2.is_alive()):\r\n        pass\r\n    assert None == t1.err == t2.err, \"Errors when run in parallel\"\r\n```"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315495913",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315495913",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315495913,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NTkxMw==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:24:26Z",
    "updated_at": "2017-07-15T00:24:26Z",
    "author_association": "MEMBER",
    "body": "> Neither of the threads throw an exception in that case.\r\n\r\nI'm confused...isn't that what you want (because it doesn't fail)?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315496144",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315496144",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315496144,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NjE0NA==",
    "user": {
      "login": "3553x",
      "id": 25621909,
      "node_id": "MDQ6VXNlcjI1NjIxOTA5",
      "avatar_url": "https://avatars0.githubusercontent.com/u/25621909?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/3553x",
      "html_url": "https://github.com/3553x",
      "followers_url": "https://api.github.com/users/3553x/followers",
      "following_url": "https://api.github.com/users/3553x/following{/other_user}",
      "gists_url": "https://api.github.com/users/3553x/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/3553x/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/3553x/subscriptions",
      "organizations_url": "https://api.github.com/users/3553x/orgs",
      "repos_url": "https://api.github.com/users/3553x/repos",
      "events_url": "https://api.github.com/users/3553x/events{/privacy}",
      "received_events_url": "https://api.github.com/users/3553x/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:27:53Z",
    "updated_at": "2017-07-15T00:27:53Z",
    "author_association": "CONTRIBUTOR",
    "body": "The threads don't throw exceptions regardless of whether I include my fix or not. Meaning the test I wrote is useless and would indicate that everything is fine while actual programs are crashing."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315496248",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315496248",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315496248,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NjI0OA==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:29:11Z",
    "updated_at": "2017-07-15T00:29:11Z",
    "author_association": "MEMBER",
    "body": "> The threads don't throw exceptions regardless of whether I include my fix or not.\r\n\r\nAh, okay.  That makes sense.  Does your code from the issue raise an Exception if you incorporate it as a test (just as a starter)?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315496777",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315496777",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315496777,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5Njc3Nw==",
    "user": {
      "login": "3553x",
      "id": 25621909,
      "node_id": "MDQ6VXNlcjI1NjIxOTA5",
      "avatar_url": "https://avatars0.githubusercontent.com/u/25621909?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/3553x",
      "html_url": "https://github.com/3553x",
      "followers_url": "https://api.github.com/users/3553x/followers",
      "following_url": "https://api.github.com/users/3553x/following{/other_user}",
      "gists_url": "https://api.github.com/users/3553x/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/3553x/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/3553x/subscriptions",
      "organizations_url": "https://api.github.com/users/3553x/orgs",
      "repos_url": "https://api.github.com/users/3553x/repos",
      "events_url": "https://api.github.com/users/3553x/events{/privacy}",
      "received_events_url": "https://api.github.com/users/3553x/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:36:20Z",
    "updated_at": "2017-07-15T00:36:20Z",
    "author_association": "CONTRIBUTOR",
    "body": "I'm not sure what you mean by starter. I tried using the code from the issue directly earlier and it didn't work. Apparently exceptions need to be thrown by the main-thread."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315497069",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315497069",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315497069,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NzA2OQ==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:40:59Z",
    "updated_at": "2017-07-15T00:40:59Z",
    "author_association": "MEMBER",
    "body": "By starter, I meant: does your code from the original issue form a valid test for our code-base (when incorporated into the `tests` directory?  You indeed answered the question."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315497254",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315497254",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315497254,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NzI1NA==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:44:09Z",
    "updated_at": "2017-07-15T00:44:25Z",
    "author_association": "MEMBER",
    "body": "@3553x : As I'm not sure at this point why `pytest` is behaving the way it is for you, how about this: you need the test to break and then pass (with your changes) on the build machines before we merge.  On a separate branch, add your test, push it, and have Travis and/or Appveyor (either one is sufficient) run and see if it breaks *without your changes*.  Then push your changes to that branch and see if they then pass.\r\n\r\nIf it does, commit your test and push to this PR.  Otherwise, let us know if it worked or not.  How does that sound to you?"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315497440",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315497440",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315497440,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NzQ0MA==",
    "user": {
      "login": "3553x",
      "id": 25621909,
      "node_id": "MDQ6VXNlcjI1NjIxOTA5",
      "avatar_url": "https://avatars0.githubusercontent.com/u/25621909?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/3553x",
      "html_url": "https://github.com/3553x",
      "followers_url": "https://api.github.com/users/3553x/followers",
      "following_url": "https://api.github.com/users/3553x/following{/other_user}",
      "gists_url": "https://api.github.com/users/3553x/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/3553x/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/3553x/subscriptions",
      "organizations_url": "https://api.github.com/users/3553x/orgs",
      "repos_url": "https://api.github.com/users/3553x/repos",
      "events_url": "https://api.github.com/users/3553x/events{/privacy}",
      "received_events_url": "https://api.github.com/users/3553x/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:46:31Z",
    "updated_at": "2017-07-15T00:46:31Z",
    "author_association": "CONTRIBUTOR",
    "body": "That sounds like something worth trying out. I'll let you know once I'm done."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315497546",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315497546",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315497546,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTQ5NzU0Ng==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T00:47:51Z",
    "updated_at": "2017-07-15T00:47:58Z",
    "author_association": "MEMBER",
    "body": "One other thing, if this operation is successful, if you could provide us to the links to the failing and passing builds.  That way we can see the differential between having and not having your patch."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315547205",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315547205",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315547205,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTU0NzIwNQ==",
    "user": {
      "login": "3553x",
      "id": 25621909,
      "node_id": "MDQ6VXNlcjI1NjIxOTA5",
      "avatar_url": "https://avatars0.githubusercontent.com/u/25621909?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/3553x",
      "html_url": "https://github.com/3553x",
      "followers_url": "https://api.github.com/users/3553x/followers",
      "following_url": "https://api.github.com/users/3553x/following{/other_user}",
      "gists_url": "https://api.github.com/users/3553x/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/3553x/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/3553x/subscriptions",
      "organizations_url": "https://api.github.com/users/3553x/orgs",
      "repos_url": "https://api.github.com/users/3553x/repos",
      "events_url": "https://api.github.com/users/3553x/events{/privacy}",
      "received_events_url": "https://api.github.com/users/3553x/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T16:57:26Z",
    "updated_at": "2017-07-15T16:57:26Z",
    "author_association": "CONTRIBUTOR",
    "body": "I tried using travis with my test added on top of the unfixed version. Some builds fail because of linting, however they all seem to pass pytest. I realised that my earlier version might have passed since `_IMPORTS` is set to true after the first call to `read_html()`. So I imported `pandas.io.html` and reloaded it. But this didn't fix the issue. This change was included in the version that was tested by travis.\r\n\r\nThese are the additions that were made to `test_html.py`:\r\n```python\r\ntry:\r\n    from importlib import reload\r\nexcept ImportError:\r\n    pass\r\n[...]\r\nimport pandas.io.html\r\n[...]\r\nclass ErrorThread(threading.Thread):\r\n    def run(self):\r\n        try:\r\n            super(ErrorThread, self).run()\r\n        except Exception as e:\r\n            self.err = e\r\n        else:\r\n            self.err = None\r\n\r\n@pytest.mark.slow\r\ndef test_importcheck_thread_safety():\r\n    reload(pandas.io.html)\r\n    filename = os.path.join(DATA_PATH, 'valid_markup.html')\r\n    helper_thread1 = ErrorThread(target = read_html, args = (filename,))\r\n    helper_thread2 = ErrorThread(target = read_html, args = (filename,))\r\n    helper_thread1.start()\r\n    helper_thread2.start()\r\n    while(helper_thread1.is_alive() or helper_thread2.is_alive()):\r\n        pass\r\n    assert None == helper_thread1.err == helper_thread2.err\r\n```\r\nOutput of `pytest test_html.py::test_importcheck_thread_safety`:\r\n```\r\n=========================== test session starts ============================\r\nplatform linux -- Python 3.6.1, pytest-3.1.1, py-1.4.34, pluggy-0.4.0\r\nrootdir: /home/fe/code/pandas, inifile: setup.cfg\r\nplugins: xdist-1.16.0\r\ncollected 8 items \r\n\r\ntest_html.py F\r\n\r\n================================= FAILURES =================================\r\n______________________ test_importcheck_thread_safety ______________________\r\n\r\n    @pytest.mark.slow\r\n    def test_importcheck_thread_safety():\r\n        reload(pandas.io.html)\r\n        filename = os.path.join(DATA_PATH, 'valid_markup.html')\r\n        helper_thread1 = ErrorThread(target = read_html, args = (filename,))\r\n        helper_thread2 = ErrorThread(target = read_html, args = (filename,))\r\n        helper_thread1.start()\r\n        helper_thread2.start()\r\n        while(helper_thread1.is_alive() or helper_thread2.is_alive()):\r\n            pass\r\n>       assert None == helper_thread1.err == helper_thread2.err\r\nE       AssertionError: assert None == ImportError('lxml not found, please install it',)\r\nE        +  where None = <ErrorThread(Thread-1, stopped 140146488710912)>.err\r\nE        +  and   ImportError('lxml not found, please install it',) = <ErrorThread(Thread-2, stopped 140146480318208)>.err\r\n\r\ntest_html.py:961: AssertionError\r\n========================= 1 failed in 7.14 seconds =========================\r\n```\r\nOutput of `pytest test_html.py`\r\n```\r\n=========================== test session starts ============================\r\nplatform linux -- Python 3.6.1, pytest-3.1.1, py-1.4.34, pluggy-0.4.0\r\nrootdir: /home/fe/code/pandas, inifile: setup.cfg\r\nplugins: xdist-1.16.0\r\ncollected 74 items \r\n\r\ntest_html.py ..........................................................................\r\n\r\n======================== 74 passed in 50.77 seconds ========================\r\n```"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315547883",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315547883",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315547883,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTU0Nzg4Mw==",
    "user": {
      "login": "jreback",
      "id": 953992,
      "node_id": "MDQ6VXNlcjk1Mzk5Mg==",
      "avatar_url": "https://avatars2.githubusercontent.com/u/953992?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jreback",
      "html_url": "https://github.com/jreback",
      "followers_url": "https://api.github.com/users/jreback/followers",
      "following_url": "https://api.github.com/users/jreback/following{/other_user}",
      "gists_url": "https://api.github.com/users/jreback/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jreback/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jreback/subscriptions",
      "organizations_url": "https://api.github.com/users/jreback/orgs",
      "repos_url": "https://api.github.com/users/jreback/repos",
      "events_url": "https://api.github.com/users/jreback/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jreback/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-15T17:06:11Z",
    "updated_at": "2017-07-15T17:06:11Z",
    "author_association": "CONTRIBUTOR",
    "body": "@3553x you can add your test to the PR."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315615679",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315615679",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315615679,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTYxNTY3OQ==",
    "user": {
      "login": "pep8speaks",
      "id": 24736507,
      "node_id": "MDQ6VXNlcjI0NzM2NTA3",
      "avatar_url": "https://avatars1.githubusercontent.com/u/24736507?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/pep8speaks",
      "html_url": "https://github.com/pep8speaks",
      "followers_url": "https://api.github.com/users/pep8speaks/followers",
      "following_url": "https://api.github.com/users/pep8speaks/following{/other_user}",
      "gists_url": "https://api.github.com/users/pep8speaks/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/pep8speaks/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/pep8speaks/subscriptions",
      "organizations_url": "https://api.github.com/users/pep8speaks/orgs",
      "repos_url": "https://api.github.com/users/pep8speaks/repos",
      "events_url": "https://api.github.com/users/pep8speaks/events{/privacy}",
      "received_events_url": "https://api.github.com/users/pep8speaks/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-16T15:12:02Z",
    "updated_at": "2017-07-21T14:55:34Z",
    "author_association": "NONE",
    "body": "Hello @3553x! Thanks for updating the PR.\n\n\n\n\n\nCheers ! There are no PEP8 issues in this Pull Request. :beers: \n\n##### Comment last updated on July 21, 2017 at 14:55 Hours UTC"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/315615844",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-315615844",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 315615844,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNTYxNTg0NA==",
    "user": {
      "login": "jreback",
      "id": 953992,
      "node_id": "MDQ6VXNlcjk1Mzk5Mg==",
      "avatar_url": "https://avatars2.githubusercontent.com/u/953992?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jreback",
      "html_url": "https://github.com/jreback",
      "followers_url": "https://api.github.com/users/jreback/followers",
      "following_url": "https://api.github.com/users/jreback/following{/other_user}",
      "gists_url": "https://api.github.com/users/jreback/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jreback/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jreback/subscriptions",
      "organizations_url": "https://api.github.com/users/jreback/orgs",
      "repos_url": "https://api.github.com/users/jreback/repos",
      "events_url": "https://api.github.com/users/jreback/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jreback/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-16T15:14:46Z",
    "updated_at": "2017-07-16T15:14:46Z",
    "author_association": "CONTRIBUTOR",
    "body": "pls rebase on master\r\n\r\n```\r\ngit rebase -i upstream/master\r\ngit push yourremote yourbranch -f\r\n```"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/316898854",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-316898854",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 316898854,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNjg5ODg1NA==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-21T04:22:17Z",
    "updated_at": "2017-07-21T04:22:17Z",
    "author_association": "MEMBER",
    "body": "@3553x : @jreback has some minor comments that if you could address, we will be able to merge this!"
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/317039904",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-317039904",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 317039904,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNzAzOTkwNA==",
    "user": {
      "login": "gfyoung",
      "id": 9273653,
      "node_id": "MDQ6VXNlcjkyNzM2NTM=",
      "avatar_url": "https://avatars0.githubusercontent.com/u/9273653?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/gfyoung",
      "html_url": "https://github.com/gfyoung",
      "followers_url": "https://api.github.com/users/gfyoung/followers",
      "following_url": "https://api.github.com/users/gfyoung/following{/other_user}",
      "gists_url": "https://api.github.com/users/gfyoung/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/gfyoung/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/gfyoung/subscriptions",
      "organizations_url": "https://api.github.com/users/gfyoung/orgs",
      "repos_url": "https://api.github.com/users/gfyoung/repos",
      "events_url": "https://api.github.com/users/gfyoung/events{/privacy}",
      "received_events_url": "https://api.github.com/users/gfyoung/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-21T15:57:28Z",
    "updated_at": "2017-07-21T15:57:36Z",
    "author_association": "MEMBER",
    "body": "@3553x : Thanks for updating!\r\n\r\n@jreback : this LGTM, but I'll leave it to you to check one more time before merging."
  },
  {
    "url": "https://api.github.com/repos/pandas-dev/pandas/issues/comments/317134308",
    "html_url": "https://github.com/pandas-dev/pandas/pull/16930#issuecomment-317134308",
    "issue_url": "https://api.github.com/repos/pandas-dev/pandas/issues/16930",
    "id": 317134308,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxNzEzNDMwOA==",
    "user": {
      "login": "jreback",
      "id": 953992,
      "node_id": "MDQ6VXNlcjk1Mzk5Mg==",
      "avatar_url": "https://avatars2.githubusercontent.com/u/953992?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jreback",
      "html_url": "https://github.com/jreback",
      "followers_url": "https://api.github.com/users/jreback/followers",
      "following_url": "https://api.github.com/users/jreback/following{/other_user}",
      "gists_url": "https://api.github.com/users/jreback/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jreback/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jreback/subscriptions",
      "organizations_url": "https://api.github.com/users/jreback/orgs",
      "repos_url": "https://api.github.com/users/jreback/repos",
      "events_url": "https://api.github.com/users/jreback/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jreback/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-07-21T23:25:15Z",
    "updated_at": "2017-07-21T23:25:15Z",
    "author_association": "CONTRIBUTOR",
    "body": "thanks @3553x "
  }
]
